src/core/qgslocalizeddatapathregistry.cpp

```diff
@@ -0,0 +1,109 @@
+/***************************************************************************
+  qgslocalizeddatapathregistry.cpp
+  --------------------------------------
+  Date                 : May 2020
+  Copyright            : (C) 2020 by Denis Rouzaud
+  Email                : denis.rouzaud
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include <QDir>
+
+#include "qgslocalizeddatapathregistry.h"
+#include "qgssettings.h"
+#include "qgis.h"
+#include "qgsreadwritelocker.h"
+
+
+QgsLocalizedDataPathRegistry::QgsLocalizedDataPathRegistry()
+{
+  readFromSettings();
+}
+
+QString QgsLocalizedDataPathRegistry::fullPath( const QString &relativePath ) const
```

Comment:
the fullPath might be relative if the project is within the localized path for instance.
Actually, I could rename fullPath to standardizedPath and relative path localizedPath.

Reply:
`localizedPath` instead of `relativePath` sounds super then, better distinguish from the standard relative path meaning. Maybe `globalizedPath` instead of `standardizedPath` too?